### DISCUSSION TITLE
TailwindCSS doesn't work in JSX elements but does using @Layer base. NextJS

### DISCUSSION THREAD

**Discussion by rober-m** [tailwindlabs/tailwindcss#6863] [Help] [answered]
Adding TailwindCSS to [this NextJS template](https://github.com/wutali/nextjs-netlify-blog-template) following [the official docs](https://tailwindcss.com/docs/guides/nextjs) I encountered a weird situation.

If I add some global config in the global.css file, something like:

```
@layer base {
	@font-face {
		font-family: 'Ubuntu';
		src: url(https://fonts.googleapis.com/css2?family=Ubuntu:wght@300;400;500&display=swap)
			format('swap');
	}
	@font-face {
		font-family: 'Ubuntu Mono';
		src: url(https://fonts.googleapis.com/css2?family=Ubuntu+Mono&display=swap)
			format('swap');
	}
	html,
	body {
		@apply bg-gray-50 text-gray-800;
		@apply font-ubuntu;
	}
	h1 {
		@apply text-4xl leading-10 mt-7;
		@apply text-green-500;
	}
	h2 {
		@apply text-3xl leading-10 mt-7;
		@apply before:content-['#'] before:-ml-5 before:mr-2 before:text-gray-400;
	}
	h3 {
		@apply text-xl leading-7 mt-3;
	}
}
```
It works perfectly.

But if I use TailwindCSS inside the JSX elements:

```html
		<div className='bg-red-500 text-3xl'>
			<h1>
			      Hi, We're Next.js & Netlify<span className='fancy'>.</span>
			</h1>
			<span className='handle'>@nextjs-netlify-blog</span>
			<h2>A blog template with Next.js and Netlify.</h2>
		</div>
```
It doesn't.

**What could be the reason? Thanks in advance**

**Comment by NovusOpus** [comment, marked as answer]
Have you configured the path of your .jsx files correctly in the `content` setting inside your `tailwind.config.js`?

Updated: 
Based on the folder structure of the repo you provided, This should work:

```JS
module.exports = {
  content: [
    "./public/**/*.{html,js,ts,jsx,tsx}",
    "./src/pages/**/*.{js,ts,jsx,tsx}",
    "./src/components/**/*.{js,ts,jsx,tsx}",
    "./src/lib/**/*.{js,ts,jsx,tsx}",
  ],
  theme: {
    extend: {},
  },
  plugins: [],
}
```
You didn't include your `public` folder path, so the Tailwind CSS classes in the templates(html, js, jsx files) are going to be ignored.
Also the correct path for the `pages` and `components` directories is:

```js
// Correct path
"./src/pages/**/*.{js,ts,jsx,tsx}",
"./src/components/**/*.{js,ts,jsx,tsx}",

// Incorrect path
'./pages/**/*.{js,ts,jsx,tsx}',
'./components/**/*.{js,ts,jsx,tsx}',
```
there's no `pages` or `components` folders in your project root(Where tailwind.config.js file is located), those folders are located inside `src`. So use the full code of the first example, that should work. 

 `content` works by scanning all of your HTML, JavaScript components, and any other template files listed in your `content` section of your `tailwind.config.js` looking for class names used, then generating all of the corresponding CSS for those styles.

You can read more about Configuring the content paths in the [Tailwind CSS Docs - Content Configuration](https://tailwindcss.com/docs/content-configuration#configuring-source-paths).

**Reply by rober-m** [reply]
Seems like that's not the issue. I Initially had it configured without the `"./src/lib/**..."` line because there are only functions in that folder. I tried adding it but it did not change the result. I also tried re-installing everything again and the problem persists. [Here's the code](https://github.com/IIPatternII/blog_nextjs/tree/reinstall_tailwind). Thank you for your help.

**Reply by NovusOpus** [reply]
@IIPatternII I updated the Original reply, You didn't added the `public` folder/path into your content section, So any Tailwind CSS Class in the files inside that path was going to be ignored. Also the correct path for the `pages` and `components` is:

```JS
"./src/pages/**/*.{js,ts,jsx,tsx}",
"./src/components/**/*.{js,ts,jsx,tsx}",
```
Since there's no `pages` or `components` folders in your project root(Where tailwind.config.js file is located), those folders are located inside `src` so your `content` is not correct, Use the code I updated in the original Reply, that should work.

**Reply by rober-m** [reply]
Thank you! It worked like a charm! Sorry I didn't notice such a clear mistake
